### ISSUE TITLE
Somatic variant caller doesn't group batches in normal/tumor with a prealigned BAM normal

### ISSUE THREAD

**Issue by lbeltrame** [bcbio/bcbio-nextgen#2356]
I've been deep debugging this but I haven't yet found the reason: probably trivial, but the batches get so much processing that it's hard to follow at a glance. Basically, the same batch with one normal with a prealigned BAM and one tumor is not properly paired, and MuTect 2 runs in tumor only mode.

Here's an excerpt from the configuration. The batch is identical, but the files aren't grouped together post-alignment (`align_bam` contains only the tumor sample).

```yaml
- algorithm:
    aligner: false
    effects: vep
    effects_transcripts: canonical
    ensemble:
      numpass: 1
      use_filtered: false
    mark_duplicates: true
    min_allele_fraction: 1
    mixup_check: qsignature
    platform: illumina
    quality_format: Standard
    realign: false
    recalibrate: false
    remove_lcr: true
    sv_regions: my_svregions
    svcaller:
    - cnvkit
    - manta
    - lumpy
    tools_off: upload_alignment
    tools_on:
    - svplots
    - bnd-genotype
    trim_reads: false
    variant_regions: my_regions
    variantcaller: false
  analysis: variant2
  description: '21175'
  files:
  - 21175-ready.bam
  genome_build: hg19
  metadata:
    batch: 21175_tumor_vs_normal
    phenotype: normal
    sex: female
 algorithm:
    aligner: bwa
    background: structural
    effects: vep
    effects_transcripts: canonical
    ensemble:
      numpass: 1
      use_filtered: false
    mark_duplicates: true
    min_allele_fraction: 1
    mixup_check: qsignature
    platform: illumina
    quality_format: Standard
    realign: false
    recalibrate: gatk
    remove_lcr: true
    sv_regions: my_svregions
    svcaller:
    - cnvkit
    - manta
    - lumpy
    tools_on:
    - svplots
    - bnd-genotype
    trim_reads: false
    variant_regions: my_regions
    variantcaller:
      somatic:
      - mutect2
  analysis: variant2
  description: 21175_tumor
  files:
  - 21175_S6_R1_001.fastq.gz
  - 21175_S6_R2_001.fastq.gz
  genome_build: hg19
  metadata:
    batch: 21175_tumor_vs_normal
    phenotype: tumor
    sex: female
```
The `structural` part is defined as:

```yaml
globals:
  my_regions: Agilent_FocusedExome.bed
  my_svregions: Agilent_OneSeq_Backbone_Covered.fixed.bed
  structural:
    cnv_reference: background-0-cnvkit.cnn
```

**Comment by lbeltrame** [comment]
I wonder if it's due to https://github.com/bcbio/bcbio-nextgen/blob/3e6ee720659c0c7877ea3a3ca2ef72da8c1c0b8e/bcbio/variation/vcfutils.py#L54 because it's unordered or something.

**Comment by lbeltrame** [comment]
Ok, my initial hunch was wrong. Up to that point, everything is still fine. Still, the `align_bam` sent to Mutect2 only contains the tumor BAM and not the normal BAM.

**Comment by lbeltrame** [comment]
Took me a *long* time but I figured what went wrong. `variantcaller` **must** be set even in normals, even if you're not doing anything with them. Otherwise they get excluded by the selection machinery, and you only have tumor samples.

@chapmanb This may need a mention in the docs if not there already, perhaps it's my own limitation in understanding but I did not expect this to happen. 

This fixes this specific issue, I have another that MuTect won't pick the panel of normals, but one at a time. ;)

**Comment by chapmanb** [comment]
Luca -- thanks much for the detailed report and identifying the underlying issue. You're exactly right: you do need to specify `variantcaller` for the normal as well. The initial input YAML told bcbio not to variantcall on the normal so it was left out of the analyses. I tried to improve the docs to make this more clear, please let me know if there is anywhere else to adjust that would help. Sorry again about the issue and hope things work cleanly for you. Let us know if you identify the MuTect PoN issue and we can also explore.
